src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1484,23 +1480,9 @@ bool DataWriterImpl::release_payload_pool()
     assert(payload_pool_);
 
     loans_.reset();
-
-    bool result = true;
-
-    PoolConfig config = PoolConfig::from_history_attributes(history_.m_att);
-    if (is_data_sharing_compatible_)
-    {
-        // No-op
-    }
-    else
-    {
-        auto topic_pool = std::static_pointer_cast<ITopicPayloadPool>(payload_pool_);
-        result = topic_pool->release_history(config, false);
```

Comment:
This must be done to free memory resources in the pool (in case we are not releasing the last one).
See `TopicPayloadPool::release_history`.

Reply:
duly noted!! 😅